packages/cli/src/services/ownership.service.ts

```diff
@@ -49,15 +49,32 @@ export class OwnershipService {
 			| ListQuery.Workflow.WithOwnedByAndSharedWith
 			| ListQuery.Credentials.WithOwnedByAndSharedWith;
 
-		Object.assign(entity, { ownedBy: null, sharedWith: [] });
+		Object.assign(entity, {
+			ownedBy: null,
+			sharedWith: [],
+			ownedByProject: null,
+			sharedWithProjects: [],
+		});
 
-		shared?.forEach(({ user, role }) => {
+		shared?.forEach(({ user, role, project }) => {
 			const { id, email, firstName, lastName } = user;
 
 			if (role === 'credential:owner' || role === 'workflow:owner') {
 				entity.ownedBy = { id, email, firstName, lastName };
+				entity.ownedByProject = {
+					id: project.id,
+					type: project.type,
+					// TODO: confirm name with product
+					name: project.name ?? 'My n8n',
```

Comment:
When you say `query` do you mean the controller method or the database query? 

Wondering because there are [virtual columns](https://typeorm.io/decorator-reference#virtualcolumn) in TypeORM and maybe you're thinking about using those?

Reply:
I mean the database query itself, where we could use a `left join` to the project owner and use an `IFNULL` statement to use the user's first name, last name and email in case the project's name is `null`

Reply:
Like we discussed, I'll check with @valya to decide on how to populate the name for personal projects and amend this in another PR.